From ba329cba788d4f320e6fc455919233222c27a0dd Mon Sep 17 00:00:00 2001
From: Kimball Thurston <kdt3rd@gmail.com>
Date: Sun, 21 Jul 2019 23:07:01 +1200
Subject: [PATCH] Fixes #139

Removes bash-isms from the autoconf bootstrap / configure.ac files

Signed-off-by: Kimball Thurston <kdt3rd@gmail.com>
---
 IlmBase/bootstrap            |  4 ++--
 IlmBase/configure.ac         | 22 ++++++++++------------
 OpenEXR/bootstrap            |  4 ++--
 OpenEXR/configure.ac         | 26 +++++++++++++-------------
 OpenEXR_Viewers/bootstrap    |  4 ++--
 OpenEXR_Viewers/configure.ac | 15 +++++++--------
 PyIlmBase/bootstrap          |  4 ++--
 PyIlmBase/configure.ac       | 15 +++++++--------
 8 files changed, 45 insertions(+), 49 deletions(-)

diff --git a/configure.ac b/configure.ac
--- a/configure.ac
+++ b/configure.ac
@@ -76,30 +76,30 @@ AC_ARG_ENABLE(cxxstd,
                              [enable ISO c++ standard 11/14 [[default=auto]]]),
               [cxxstd="${enableval}"], [cxxstd=14])
 
-if test "${cxxstd}" == 17 ; then
+if test "${cxxstd}" = 17 ; then
     AX_CXX_COMPILE_STDCXX([17], [noext], [mandatory])
 	CXXFLAGS="$CXXFLAGS -std=c++17"
 else
-  if test "${cxxstd}" == 14 ; then
+  if test "${cxxstd}" = 14 ; then
       AX_CXX_COMPILE_STDCXX([14], [noext], [mandatory])
   	CXXFLAGS="$CXXFLAGS -std=c++14"
   else
-    if test "${cxxstd}" == 11 ; then
+    if test "${cxxstd}" = 11 ; then
       AX_CXX_COMPILE_STDCXX([11], [noext], [mandatory])
       CXXFLAGS="$CXXFLAGS -std=c++11"
     else
-      if test "${cxxstd}" == 03 ; then
+      if test "${cxxstd}" = 03 ; then
         CXXFLAGS="$CXXFLAGS -std=c++03"
       else
         dnl automatically determine...
         AX_CXX_COMPILE_STDCXX([11], [noext], [optional])
         AX_CXX_COMPILE_STDCXX([14], [noext], [optional])
         AX_CXX_COMPILE_STDCXX([17], [noext], [optional])
-        if test "$HAVE_CXX14" == 1 ; then
+        if test "$HAVE_CXX14" = 1 ; then
   	      CXXFLAGS="$CXXFLAGS -std=c++14"
           cxxstd = 14
         else
-          if test "$HAVE_CXX11" == 1 ; then
+          if test "$HAVE_CXX11" = 1 ; then
   	        CXXFLAGS="$CXXFLAGS -std=c++11"
             cxxstd = 11
           fi
@@ -216,7 +216,7 @@ AC_COMPILE_IFELSE(
    [sysconf_nproc="no"]
 )
 AC_MSG_RESULT([$sysconf_nproc])
-if test "x${sysconf_nproc}" == xyes ; then
+if test "x${sysconf_nproc}" = xyes ; then
     AC_DEFINE([OPENEXR_IMF_HAVE_SYSCONF_NPROCESSORS_ONLN], [1], [Define if sysconf(_SC_NPROCESSORS_ONLN) can be used for CPU count])
 fi
 
@@ -297,14 +297,14 @@ LIB_SUFFIX=""
 lib_suffix_valid="no"
 
 lib_namespace="Imf"
-if test "x${library_namespace_versioning}" == xyes ; then
+if test "x${library_namespace_versioning}" = xyes ; then
     AC_DEFINE_UNQUOTED([OPENEXR_IMF_INTERNAL_NAMESPACE], [Imf_${OPENEXR_VERSION_API}], [OpenEXR])
     AC_DEFINE([OPENEXR_IMF_INTERNAL_NAMESPACE_CUSTOM], [1], [OpenEXR])
 
     lib_namespace="Imf_${OPENEXR_VERSION_API}"
     LIB_SUFFIX="${OPENEXR_VERSION_API}"
     lib_suffix_valid="yes"
-elif test "x${library_namespace_versioning}" == xno ; then
+elif test "x${library_namespace_versioning}" = xno ; then
     AC_DEFINE_UNQUOTED([OPENEXR_IMF_INTERNAL_NAMESPACE], [Imf], [OpenEXR])
 
     lib_namespace="Imf"
@@ -318,7 +318,7 @@ else
 fi
 AC_SUBST(LIB_SUFFIX)
 
-if test "x${lib_suffix_valid}" == xyes ; then
+if test "x${lib_suffix_valid}" = xyes ; then
 AC_SUBST(LIB_SUFFIX_DASH,"-${LIB_SUFFIX}")
 AM_CONDITIONAL(LIB_SUFFIX_EXISTS,true)
 else
@@ -338,11 +338,11 @@ AC_ARG_ENABLE(customusernamespace,
         [custom_usr_namespace="${enableval}"], 
         [custom_usr_namespace=no])
 
-if test "x${custom_usr_namespace}" == xyes ; then
+if test "x${custom_usr_namespace}" = xyes ; then
     AC_MSG_WARN([Enabling 'custom user namespace' requires an additional argument, reverting to 'Imf'])
     AC_DEFINE_UNQUOTED([OPENEXR_IMF_NAMESPACE], [Imf], [OpenEXR])
     usr_namespace="Imf"
-elif test "x${custom_usr_namespace}" == xno ; then
+elif test "x${custom_usr_namespace}" = xno ; then
     AC_DEFINE_UNQUOTED([OPENEXR_IMF_NAMESPACE], [Imf], [OpenEXR])
     usr_namespace="Imf"
 else
@@ -386,7 +386,7 @@ enable large stack optimizations                $large_stack
 internal library namespace                      $lib_namespace
 user-client namespace                           $usr_namespace])
 
-if test "x$build_osxuniversal" == xyes; then
+if test "x$build_osxuniversal" = xyes; then
 AC_MSG_RESULT([
 build OS X universal binaries                   $build_osxuniversal])
 fi
